core/os/input_event.cpp

```diff
@@ -34,8 +34,52 @@
  */
 
 bool InputEvent::operator==(const InputEvent &p_event) const {
+	if (type != p_event.type){
+		return false;
+	}
 
-	return true;
+	switch(type) {
```

Comment:
@akien-mga I now know that how to merge two commits to one, but I don't know how to separate one commit in tow. :(